exp/hubble/schema_factory.go

```
@@ -107,48 +137,53 @@ func makeAccountIDFromChange(change *xdr.LedgerEntryChange) (string, error) {
 }
 
 func makeSeqnum(state *accountState, change *xdr.LedgerEntryChange) (uint32, error) {
-	var seqnum xdr.Uint32
-	switch entryType := change.Type; entryType {
-	case xdr.LedgerEntryChangeTypeLedgerEntryCreated:
-		seqnum = change.MustCreated().LastModifiedLedgerSeq
-	case xdr.LedgerEntryChangeTypeLedgerEntryUpdated:
-		seqnum = change.MustUpdated().LastModifiedLedgerSeq
-	case xdr.LedgerEntryChangeTypeLedgerEntryState:
-		seqnum = change.MustState().LastModifiedLedgerSeq
-
-	// We do not need to update the seqnum for removed changes, because
-	// we just remove the accompanying account's state.
-	case xdr.LedgerEntryChangeTypeLedgerEntryRemoved:
-		return 0, nil
-	default:
-		return 0, fmt.Errorf("Unknown entry type: %v", entryType)
+	// Removed entries would not be of Account type, and thus
+	// do not change the last modified ledger seqnum.
+	if change.Type == xdr.LedgerEntryChangeTypeLedgerEntryRemoved {
+		return state.seqnum, nil
+	}
+
+	entry, ok := change.GetLedgerEntry()
+	if !ok {
+		return state.seqnum, fmt.Errorf("Could not get ledger entry")
 	}
-	return uint32(seqnum), nil
+	return uint32(entry.LastModifiedLedgerSeq), nil
 }
 
 func makeBalance(state *accountState, change *xdr.LedgerEntryChange) (uint32, error) {
-	account, err := getAccountEntry(change)
-	if err != nil {
-		return 0, err
+	// If the change is a non-account removal or a change of non-account type, then
+	// we simply return the current balance (or 0, if we were passed a nil state pointer).
+	if change.Type == xdr.LedgerEntryChangeTypeLedgerEntryRemoved {
+		return state.balance, nil
+	}
+
+	entry, ok := change.GetLedgerEntry()
+	if !ok {
+		return state.balance, nil
 	}
 
-	// The change does not update the account state, so we return
-	// the current balance.
-	if account == nil {
+	account, ok := entry.Data.GetAccount()
+	if !ok {
 		return state.balance, nil
```

here too, should return an error?
if we should not return an error then we should specify why in a comment since it's not obvious to a reader